pyomo/core/base/param.py

```diff
@@ -710,8 +707,7 @@ def _initialize_from(self, _init):
             # will take the "less surprising" route of letting the
             # source become dense, so that we get the expected copy.
             #
-            # TODO: Establish use-cases where we can use default values
-            # for sparsity.
+            # TBD: Are there use-cases where we want to maintain sparsity?
```

Comment:
I think you should create a ticket to follow-up on this action.  The need for sparsity seems more obvious to me than it does to you.